### DISCUSSION TITLE
.par file specifications: dt and regularization

### DISCUSSION THREAD

**Discussion by turbulentphil** [Nek5000/nekRS#561] [Q&A] [answered]
Dear Neks,

I have been using Nek5000 for a couple of years, am trying to get started with NekRS, and struggle understanding some of the parameters in the parameter file. In case it is of any importance for specifying the below parameters: I run DNS of Rayleigh-Bénard convection.

I would like to **understand the following two specifications** under the [GENERAL] section (more elaborations below):
1) `dt = A +targetCFL=B +max=C +initial=D`
2) `regularization = hpfrt + nModes=E + scalingCoeff=F`

**To 1:** 
1.1) Assume we have specified a `dt`-value of A. How does `targetCFL` work? Will it only limit the actual dt in case of a too high CFL number? (Like in Nek500 when specifying `targetCFL=B` and `variableDT=yes` ?!)
1.2) What does `max` specify? Does it refer to `dt` or to `targetCFL`?
1.3) What does `initial` specify?
1.4) I am surprised of the extreme CFL number compared to Nek5000 where the recommended limit was `targetCFL=0.5`. What is the recommended limit for NekRS? I find that different examples apply different `targetCFL` values.

**To 2:**
2.1) What is the regularisation specified here at all? What is its use?
2.2) In which use cases is it recommended to make use of the regularisation?
2.3) What are recommended values for `nModes` and `scalingCoeff`? I find that different examples apply different values.

I am aware of the _parHelp.txt_ file, but unfortunately it doesn't (at least for me) clarify the situation sufficiently. As it is important to understand how one specifies the numerical code or simulation, I would be thankful for any help.

Best,
Philipp

**Comment by yslan** [comment, marked as answer]
**dt**
- 1.1 `dt = A + targetCFL=B +max=C +initial=D`
  When one of the tags (`targetcfl`, `max`, `initial`) is used, it turns on the variable dt. The idea is to increase / decrease dt so the CFL is approximately the targetCFL (by default, it's 0.5 if not set in par file)
- 1.2 max dt.
- 1.3 The initial value of dt
- 1.4 It's based on the stability region. max CFL = 0.6339 for BDF3/EXT3. 
   If you have more non-linearity from source term, it's recommended to be more conservative. CFL = 0.5 is a ok number. If you use OIF (characteristic with RK substeps), then you can increase the target CFL = 0.6339 * number_of_substeps. 

**regularization**
- 2.1 regularization = filter. 
- 2.2 It's recommended to have filter to make the under-resolved solution stable.
- 2.3 It totally depends on your case, Typically, nModes=1 or 2 is sufficient for polynomial order 7.  See "high-pass filter" [ref](https://nek5000.github.io/NekDoc/problem_setup/filter.html#high-pass-filter) 

Hope this helps,
Yu-Hsiang

**Reply by turbulentphil** [reply]
Dear Yu-Hsiang,
thank you for your explanations!

**dt**
1. Thanks for clarifying how `targetCFL`, `max`, and `initial` work and to what they refer.
2. On the one hand, you mention CFL numbers of 0.5 ... 0.8. On the other hand, some examples (such as rbc) use `targetCFL=4` which is 5 to 10 times as high. Why is this numbers so high, or how can it be stable? Is it only because of the spatial filtering applied by `regularization` (although it is a _spatial_ filter only)?
3. What does OIF abbreviate? And maybe you could tell me what `tombo1,2,3` abbreviate. It is obvious that the number gives the order, but what scheme is implemented?

**regularisation**
1. Thanks for referring to the Nek5000's documentation of the spatial filtering technique, that is a great help! 
2. So when we run DNS that resolve all dynamical scales, we do not need the regularisation since there is no flow at sub-grid scales.

Best,
Philipp

**Reply by yslan** [reply]
OIFS stands for operator-integration-factor splitting, also called characteristic scheme in Nek5000. 
See the [paper](https://doi.org/10.1007/s10915-018-0876-6) for the detailed scheme and also check the [chaper](https://doi.org/10.1016/B978-0-32-391144-3.00009-7) for explanation why it can achieve CFL=4. The stability is ensured by the eigenvalue studies, the stability region for a given temporal scheme, and the subcycling of the characteristic method. Filter is usually for suppressing the oscillatory behavior from under-resolved mesh which is different. 


`tombo1,2,3` are first, second and third order accurate temporal schemes used in Nek [(reference)](https://doi.org/10.1007/BF01061059). It contains the semi-explicit schemes (BDF1+AB3 / BDF2+modified EXT3 / BDF3+EXT3) and the spitting scheme (or called PNPN in Nek5000) for satisfying the divergence-free constraint. 

Yes, DNS shouldn't apply any filtering. LES would have some condition to set filter with regards to the grid spacing. This is not my specialty but you can find some existing papers of Nek5000 simulations.

**Reply by turbulentphil** [reply]
Thank you a lot! This made the situation much clearer for me.
